models/academia/branch.js

```diff
@@ -32,12 +34,24 @@ const branchSchema = new Schema({
   },
 });
 
-branchSchema.methods.getSemester = function(semesterNumber) {
-  return utility.arrayUtil.findNeedle(
-      this.semesters,
-      semesterNumber,
-      'semester',
-  );
+branchSchema.methods.getSemester = function(semesterName) {
+  return _.find(this.semesters, (semester) => {
+    return semester.semester == semesterName;
+  });
+};
+
+branchSchema.methods.getSemesterById = function(semesterId) {
+  return _.find(this.branches, (semester) => {
+    return semester.semesterId.equals(semesterId);
+  });
+};
+
+branchSchema.methods.getSemesterByFormerName = function(semesterName) {
+  return _.find(this.branches, (semester) => {
+    return _.find(semester.semesterNameHistory, (formerSemesterName) => {
+      return formerSemesterName == semesterName;
```

Comment:
either fix to `==` format or stick to regex test format as it is in other api's

Reply:
We cannot use regex for semester since semester _1_ will also match semester _11_ and _12_. So instead of matching substrings (as in course, branch and subject), in semester we need to do exact match.

Reply:
Hmm, string data type for semester makes it kinda special, then in that case I would suggest adding a comment explaining this for semester data types

Reply:
Agreed, I will add comment explaining this case.